containers/ecr-viewer/src/app/tests/services/evaluateFhirDataServices.test.ts

```diff
@@ -427,6 +435,164 @@ Home: 123-456-6909`,
+          {
+            code: "46241-6",
+            system: "http://loinc.org",
+            display: "Hospital Admission Diagnosis",
+          },
+        ],
+      },
+      mode: "snapshot",
+      entry: [
+        {
+          display:
+            "Problem - Disease caused by severe acute respiratory syndrome coronavirus 2 (disorder)",
+          reference: "Condition/d42c4a1f-f700-61bf-62a0-c034257d6a79",
+        },
+      ],
+    };
+
+    const dischargeDiagnosis = {
+      id: "e9c9e752-dfae-c13d-a4c0-64cef027435f",
+      title: "Discharge Diagnosis",
+      text: {
+        status: "generated",
+        div: "Diverticula of intestine",
+      },
+      code: {
+        coding: [
+          {
+            code: "11535-2",
+            system: "http://loinc.org",
+            display: "Hospital Discharge Diagnosis",
+          },
+        ],
+      },
+      mode: "snapshot",
+      entry: [
+        {
+          display:
+            "Problem - Disease caused by severe acute respiratory syndrome coronavirus 2 (disorder)",
+          reference: "Condition/d42c4a1f-f700-61bf-62a0-c034257d6a79",
+        },
+      ],
+    };
+
+    const addSectionsToBundle = (
+      newSections: object[],
+      bundle: Bundle,
+    ): Bundle => {
+      return {
+        ...bundle,
+        entry: [
+          {
+            ...bundle.entry![0],
+            // @ts-ignore
+            resource: {
+              ...bundle.entry![0].resource,
+              section: [
+                ...(bundle.entry![0].resource?.section || []),
+                ...newSections,
+              ],
+            },
+          },
+          ...bundle.entry!.slice(1),
+        ],
+      };
+    };
+
+    it("should return unavailable data when no Admission Diagnosis or Discharge diagnosis are found", () => {
+      const bundle: Bundle = {
+        resourceType: "Bundle",
+        type: "document",
+        entry: [],
+      };
+
+      expect(evaluateHospitalEncounterData(bundle)).toEqual({
+        availableData: [],
+        unavailableData: [
+          {
+            table: true,
+            title: "Hospital Admission Diagnosis",
+            value: undefined,
+          },
+          {
+            table: true,
+            title: "Hospital Discharge Diagnosis",
+            value: undefined,
+          },
+        ],
+      });
+    });
+
+    it("should return Hospital Encounter Data when present and match snapshot", () => {
+      // Create a bundle with Admission and Discharge Dx
+      const bundleWithHospitalEncounterData = addSectionsToBundle(
+        [admissionDiagnosis, dischargeDiagnosis],
+        BundlePatientWithCovid,
+      );
+
+      const actual = evaluateHospitalEncounterData(
+        bundleWithHospitalEncounterData,
+      );
+
+      expect(actual).toMatchSnapshot();
+      expect(actual.availableData[0].title).toEqual(
+        "Hospital Admission Diagnosis",
+      );
+      expect(actual.availableData[1].title).toEqual(
```

Comment:
can we check the diagnosis text here as well? make sure we're pulling out the code/datetime correctly